scripts/runtests.py

```diff
@@ -0,0 +1,149 @@
+        print("Could not run tests due to an internal error with Black. See above for details.", file=sys.stderr)
+        sys.exit(black_result.returncode)
+
+    print("\nRunning Flake8...")
+    flake8_result = subprocess.run([sys.executable, "-m", "flake8", path])
+
+    print("\nRunning check_consistent.py...")
+    check_consistent_result = subprocess.run([sys.executable, "tests/check_consistent.py"])
+    print("\nRunning check_new_syntax.py...")
+    check_new_syntax_result = subprocess.run([sys.executable, "tests/check_new_syntax.py"])
+
+    print("\nRunning Pyright...")
+    pyright_result = subprocess.run(
+        [sys.executable, "tests/pyright_test.py", path] + _get_strict_params(path), stderr=subprocess.PIPE, text=True
+    )
+    print(pyright_result.stderr)
+    if re.match(r"error (runn|find)ing npx", pyright_result.stderr):
+        pyright_returncode = 0
+        pyright_skipped = True
+    else:
+        pyright_returncode = pyright_result.returncode
+        pyright_skipped = False
+
+    print("\nRunning mypy...")
+    mypy_result = subprocess.run([sys.executable, "tests/mypy_test.py", path])
+    # If mypy failed, stubtest will fail without any helpful error
+    if mypy_result.returncode == 0:
+        print("\nRunning stubtest...")
+        if folder == "stdlib":
+            stubtest_result = subprocess.run([sys.executable, "tests/stubtest_stdlib.py", stub])
+        else:
+            stubtest_result = subprocess.run([sys.executable, "tests/stubtest_third_party.py", stub])
+    else:
+        print("Skipping stubtest since mypy failed.")
+
+    if sys.platform == "win32":
+        print("Skipping pytype on Windows. You can run the test with WSL.")
+    else:
+        print("\nRunning pytype...")
+        pytype_result = subprocess.run([sys.executable, "tests/pytype_test.py", path])
+
+    print("\nRunning regression tests...")
+    regr_test_result = subprocess.run(
+        [sys.executable, "tests/regr_test.py", "stdlib" if folder == "stdlib" else stub], stderr=subprocess.PIPE, text=True
+    )
+    # No test means they all ran successfully (0 out of 0). Not all 3rd-party stubs have regression tests.
+    if "No test cases found" in regr_test_result.stderr:
+        regr_test_returncode = 0
+        print(f"No test cases found for '{stub}'!")
+    else:
+        regr_test_returncode = regr_test_result.returncode
+        print(regr_test_result.stderr)
+
+    any_failure = any(
+        [
+            flake8_result.returncode,
+            check_consistent_result.returncode,
+            check_new_syntax_result.returncode,
+            pyright_returncode,
+            mypy_result.returncode,
+            getattr(stubtest_result, "returncode", 0),
+            getattr(pytype_result, "returncode", 0),
+            regr_test_returncode,
+        ]
+    )
+
+    if any_failure:
+        print("\nOne or more tests failed. See above for details.")
+    else:
+        print("\nAll tests passed!")
```

Comment:
Makes it a little more obvious where the output from the various scripts ends, and where the summary of all the tests begins.

```suggestion
    if any_failure:
        print(colored("\n\n--- TEST SUMMARY: One or more tests failed. See above for details. ---\n\n", "red"))
    else:
        print(colored("\n\n--- TEST SUMMARY: All tests passed! ---\n\n", "green"))
```